eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/runner/ClusteringCommandLineRunner.java

```diff
@@ -0,0 +1,150 @@
+import org.springframework.boot.autoconfigure.batch.JobLauncherCommandLineRunner;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.stereotype.Component;
+import org.springframework.util.PatternMatchUtils;
+import uk.ac.ebi.eva.accession.clustering.parameters.InputParameters;
+import uk.ac.ebi.eva.commons.batch.exception.NoJobToExecuteException;
+import uk.ac.ebi.eva.commons.batch.exception.NoParametersHaveBeenPassedException;
+import uk.ac.ebi.eva.commons.batch.exception.NoPreviousJobExecutionException;
+import uk.ac.ebi.eva.commons.batch.exception.UnknownJobException;
+import uk.ac.ebi.eva.commons.batch.job.JobExecutionApplicationListener;
+import uk.ac.ebi.eva.commons.batch.job.JobStatusManager;
+
+import java.util.Collection;
+
+@Component
+public class ClusteringCommandLineRunner extends JobLauncherCommandLineRunner implements
+        ApplicationEventPublisherAware, ExitCodeGenerator {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClusteringCommandLineRunner.class);
+
+    public static final String SPRING_BATCH_JOB_NAME_PROPERTY = "spring.batch.job.names";
+
+    public static final int EXIT_WITHOUT_ERRORS = 0;
+
+    public static final int EXIT_WITH_ERRORS = 1;
+
+    @Value("${" + SPRING_BATCH_JOB_NAME_PROPERTY + ":#{null}}")
+    private String jobName;
+
+    private Collection<Job> jobs;
+
+    private JobRepository jobRepository;
+
+    @Autowired
+    private JobExecutionApplicationListener jobExecutionApplicationListener;
+
+    @Autowired
+    private InputParameters inputParameters;
+
+    private boolean abnormalExit;
+
+    public ClusteringCommandLineRunner(JobLauncher jobLauncher, JobExplorer jobExplorer,
+                                       JobRepository jobRepository) {
+        super(jobLauncher, jobExplorer);
+        this.jobRepository = jobRepository;
+    }
+
+    @Autowired(required = false)
+    public void setJobs(Collection<Job> jobs) {
+        this.jobs = jobs;
+    }
+
+    @Override
+    public void setJobNames(String jobName) {
+        this.jobName = jobName;
+        super.setJobNames(jobName);
+    }
+
+    @Override
+    public int getExitCode() {
+        if (!abnormalExit && jobExecutionApplicationListener.isJobExecutionComplete()) {
+            return EXIT_WITHOUT_ERRORS;
+        } else {
+            return EXIT_WITH_ERRORS;
+        }
+    }
+
+    @Override
+    public void run(String... args) throws JobExecutionException {
+        try {
+            abnormalExit = false;
+
+            // TODO: different jobs can have different parameters
+            JobParameters jobParameters = inputParameters.toJobParameters();
+
+            JobStatusManager.checkIfJobNameHasBeenDefined(jobName);
+            JobStatusManager.checkIfPropertiesHaveBeenProvided(jobParameters);
+            if (inputParameters.isForceRestart()) {
+                markPreviousJobAsFailed(jobParameters);
+            }
```

Comment:
if this PR is merged after https://github.com/EBIvariation/eva-accession/pull/204, look at this region in other modules as sundar made some changes regarding how spring handles job identification.

Reply:
For now, I will hold off on merging PR#204 until this PR is merged since I will have to consider these new changes in the eva-accession-clustering module for Spring 2 upgrade as well.